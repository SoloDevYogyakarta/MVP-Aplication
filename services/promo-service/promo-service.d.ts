/// <reference types="multer" />
import { HttpStatus } from '@nestjs/common';
import { PromoField } from '../../dto/promo-dto/promo-dto';
export declare class PromoService {
    private readonly logger;
    create(user_id: number, body: PromoField, file: Express.Multer.File): Promise<{
        status: HttpStatus;
        message: string;
        create: import("../../database/entities/services/promo-entity/promo-entity").PromoInstance;
    }>;
    update(id: number, body: PromoField, file: Express.Multer.File): Promise<{
        status: HttpStatus;
        message: string;
    }>;
    destroy(id: number): Promise<{
        status: HttpStatus;
        message: string;
    }>;
}
