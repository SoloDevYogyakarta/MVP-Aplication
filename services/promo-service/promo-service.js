"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var PromoService_1;
Object.defineProperty(exports, "__esModule", { value: true });
exports.PromoService = void 0;
const common_1 = require("@nestjs/common");
const system_1 = require("../../utils/system/system");
const promo_entity_1 = require("../../database/entities/services/promo-entity/promo-entity");
let PromoService = PromoService_1 = class PromoService {
    constructor() {
        this.logger = new common_1.Logger(PromoService_1.name);
    }
    async create(user_id, body, file) {
        this.logger.log(PromoService_1.name);
        let image = null;
        const findOne = await promo_entity_1.promoEntity.findOne({ where: { name: body.name } });
        if (findOne) {
            throw new common_1.HttpException({
                status: common_1.HttpStatus.BAD_REQUEST,
                message: 'Promo name already exists',
            }, common_1.HttpStatus.BAD_REQUEST);
        }
        if (file?.path) {
            image = `/assets${file.path.split('/assets')[1]}`;
        }
        const create = await promo_entity_1.promoEntity.create({ ...body, image, user_id });
        create.save();
        return {
            status: common_1.HttpStatus.CREATED,
            message: 'Promo has been create',
            create,
        };
    }
    async update(id, body, file) {
        this.logger.log(PromoService_1.name);
        const findOne = await promo_entity_1.promoEntity.findOne({ where: { id } });
        if (!findOne) {
            throw new common_1.HttpException({
                status: common_1.HttpStatus.BAD_REQUEST,
                message: 'Promo not found',
            }, common_1.HttpStatus.BAD_REQUEST);
        }
        if (file?.path) {
            const image = `/assets${file.path.split('/assets')[1]}`;
            if (findOne.image) {
                try {
                    (0, system_1.removepath)(`../..${findOne.image}`);
                }
                catch (err) {
                }
            }
            findOne.image = image;
            findOne.save();
        }
        findOne.update(body, { where: { id } });
        return { status: common_1.HttpStatus.OK, message: 'Promo has been update' };
    }
    async destroy(id) {
        this.logger.log(PromoService_1.name);
        const findOne = await promo_entity_1.promoEntity.findOne({ where: { id } });
        if (!findOne) {
            throw new common_1.HttpException({
                status: common_1.HttpStatus.BAD_REQUEST,
                message: 'Promo not found',
            }, common_1.HttpStatus.BAD_REQUEST);
        }
        if (findOne.image) {
            try {
                (0, system_1.removepath)(`../..${findOne.image}`);
            }
            catch (err) {
            }
        }
        findOne.destroy();
        return { status: common_1.HttpStatus.OK, message: 'Promo has been delete' };
    }
};
exports.PromoService = PromoService;
exports.PromoService = PromoService = PromoService_1 = __decorate([
    (0, common_1.Injectable)()
], PromoService);
//# sourceMappingURL=promo-service.js.map