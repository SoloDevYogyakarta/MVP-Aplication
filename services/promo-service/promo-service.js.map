{"version":3,"file":"promo-service.js","sourceRoot":"","sources":["../../../src/services/promo-service/promo-service.ts"],"names":[],"mappings":";;;;;;;;;;AAAA,2CAA+E;AAC/E,sDAAuD;AACvD,6FAAyF;AAIlF,IAAM,YAAY,oBAAlB,MAAM,YAAY;IAAlB;QACY,WAAM,GAAG,IAAI,eAAM,CAAC,cAAY,CAAC,IAAI,CAAC,CAAC;IA6E1D,CAAC;IA3EC,KAAK,CAAC,MAAM,CAAC,OAAe,EAAE,IAAgB,EAAE,IAAyB;QACvE,IAAI,CAAC,MAAM,CAAC,GAAG,CAAC,cAAY,CAAC,IAAI,CAAC,CAAC;QACnC,IAAI,KAAK,GAAG,IAAI,CAAC;QACjB,MAAM,OAAO,GAAG,MAAM,0BAAW,CAAC,OAAO,CAAC,EAAE,KAAK,EAAE,EAAE,IAAI,EAAE,IAAI,CAAC,IAAI,EAAE,EAAE,CAAC,CAAC;QAC1E,IAAI,OAAO,EAAE;YACX,MAAM,IAAI,sBAAa,CACrB;gBACE,MAAM,EAAE,mBAAU,CAAC,WAAW;gBAC9B,OAAO,EAAE,2BAA2B;aACrC,EACD,mBAAU,CAAC,WAAW,CACvB,CAAC;SACH;QACD,IAAI,IAAI,EAAE,IAAI,EAAE;YACd,KAAK,GAAG,UAAU,IAAI,CAAC,IAAI,CAAC,KAAK,CAAC,SAAS,CAAC,CAAC,CAAC,CAAC,EAAE,CAAC;SACnD;QACD,MAAM,MAAM,GAAG,MAAM,0BAAW,CAAC,MAAM,CAAC,EAAE,GAAG,IAAI,EAAE,KAAK,EAAE,OAAO,EAAE,CAAC,CAAC;QACrE,MAAM,CAAC,IAAI,EAAE,CAAC;QACd,OAAO;YACL,MAAM,EAAE,mBAAU,CAAC,OAAO;YAC1B,OAAO,EAAE,uBAAuB;YAChC,MAAM;SACP,CAAC;IACJ,CAAC;IAED,KAAK,CAAC,MAAM,CAAC,EAAU,EAAE,IAAgB,EAAE,IAAyB;QAClE,IAAI,CAAC,MAAM,CAAC,GAAG,CAAC,cAAY,CAAC,IAAI,CAAC,CAAC;QACnC,MAAM,OAAO,GAAG,MAAM,0BAAW,CAAC,OAAO,CAAC,EAAE,KAAK,EAAE,EAAE,EAAE,EAAE,EAAE,CAAC,CAAC;QAC7D,IAAI,CAAC,OAAO,EAAE;YACZ,MAAM,IAAI,sBAAa,CACrB;gBACE,MAAM,EAAE,mBAAU,CAAC,WAAW;gBAC9B,OAAO,EAAE,iBAAiB;aAC3B,EACD,mBAAU,CAAC,WAAW,CACvB,CAAC;SACH;QACD,IAAI,IAAI,EAAE,IAAI,EAAE;YACd,MAAM,KAAK,GAAG,UAAU,IAAI,CAAC,IAAI,CAAC,KAAK,CAAC,SAAS,CAAC,CAAC,CAAC,CAAC,EAAE,CAAC;YACxD,IAAI,OAAO,CAAC,KAAK,EAAE;gBACjB,IAAI;oBACF,IAAA,mBAAU,EAAC,QAAQ,OAAO,CAAC,KAAK,EAAE,CAAC,CAAC;iBACrC;gBAAC,OAAO,GAAG,EAAE;iBAEb;aACF;YACD,OAAO,CAAC,KAAK,GAAG,KAAK,CAAC;YACtB,OAAO,CAAC,IAAI,EAAE,CAAC;SAChB;QACD,OAAO,CAAC,MAAM,CAAC,IAAI,EAAE,EAAE,KAAK,EAAE,EAAE,EAAE,EAAE,EAAE,CAAC,CAAC;QACxC,OAAO,EAAE,MAAM,EAAE,mBAAU,CAAC,EAAE,EAAE,OAAO,EAAE,uBAAuB,EAAE,CAAC;IACrE,CAAC;IAED,KAAK,CAAC,OAAO,CAAC,EAAU;QACtB,IAAI,CAAC,MAAM,CAAC,GAAG,CAAC,cAAY,CAAC,IAAI,CAAC,CAAC;QACnC,MAAM,OAAO,GAAG,MAAM,0BAAW,CAAC,OAAO,CAAC,EAAE,KAAK,EAAE,EAAE,EAAE,EAAE,EAAE,CAAC,CAAC;QAC7D,IAAI,CAAC,OAAO,EAAE;YACZ,MAAM,IAAI,sBAAa,CACrB;gBACE,MAAM,EAAE,mBAAU,CAAC,WAAW;gBAC9B,OAAO,EAAE,iBAAiB;aAC3B,EACD,mBAAU,CAAC,WAAW,CACvB,CAAC;SACH;QACD,IAAI,OAAO,CAAC,KAAK,EAAE;YACjB,IAAI;gBACF,IAAA,mBAAU,EAAC,QAAQ,OAAO,CAAC,KAAK,EAAE,CAAC,CAAC;aACrC;YAAC,OAAO,GAAG,EAAE;aAEb;SACF;QACD,OAAO,CAAC,OAAO,EAAE,CAAC;QAClB,OAAO,EAAE,MAAM,EAAE,mBAAU,CAAC,EAAE,EAAE,OAAO,EAAE,uBAAuB,EAAE,CAAC;IACrE,CAAC;CACF,CAAA;AA9EY,oCAAY;uBAAZ,YAAY;IADxB,IAAA,mBAAU,GAAE;GACA,YAAY,CA8ExB"}