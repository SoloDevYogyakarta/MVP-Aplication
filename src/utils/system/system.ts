import fs from 'fs';
import { join } from 'path';

function joinpath(path: string): string {
  return join(__dirname, path);
}

function readpath(path: string): string {
  return fs.readFileSync(joinpath(path), { encoding: 'utf-8' });
}

function createpath(path: string, data: unknown): void {
  let values = data;
  if (typeof values !== 'string') {
    values = JSON.stringify(values);
  }
  fs.writeFileSync(joinpath(path), values as string);
}

function removepath(path: string): void {
  fs.unlinkSync(joinpath(path));
}

export { joinpath, readpath, createpath, removepath };
