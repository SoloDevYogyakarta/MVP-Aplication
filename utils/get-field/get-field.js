"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getField = void 0;
const system_1 = require("../system/system");
function getField(filename) {
    try {
        const inc = ['token', 'fake'];
        if (inc.includes(filename)) {
            return (0, system_1.readpath)(`../../database/dataTxt/${filename}.txt`);
        }
        return JSON.parse((0, system_1.readpath)(`../../database/dataTxt/${filename}.txt`));
    }
    catch (err) {
        throw new Error(err);
    }
}
exports.getField = getField;
//# sourceMappingURL=get-field.js.map