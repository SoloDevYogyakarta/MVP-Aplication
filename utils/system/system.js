"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.removepath = exports.readpath = exports.createpath = exports.joinpath = void 0;
const fs_1 = __importDefault(require("fs"));
const path_1 = require("path");
function joinpath(path) {
    return (0, path_1.join)(__dirname, path);
}
exports.joinpath = joinpath;
function createpath(path, data) {
    if (typeof data !== 'string') {
        data = JSON.stringify(data);
    }
    fs_1.default.writeFileSync(joinpath(path), data);
}
exports.createpath = createpath;
function readpath(path) {
    return fs_1.default.readFileSync(joinpath(path), { encoding: 'utf-8' });
}
exports.readpath = readpath;
function removepath(path) {
    fs_1.default.unlinkSync(joinpath(path));
}
exports.removepath = removepath;
//# sourceMappingURL=system.js.map