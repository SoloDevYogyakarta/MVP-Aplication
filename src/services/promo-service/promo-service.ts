import { HttpException, HttpStatus, Injectable, Logger } from '@nestjs/common';
import { removepath } from '../../utils/system/system';
import { promoEntity } from '../../database/entities/services/promo-entity/promo-entity';
import { PromoField } from '../../dto/promo-dto/promo-dto';

@Injectable()
export class PromoService {
  private readonly logger = new Logger(PromoService.name);

  async create(user_id: number, body: PromoField, file: Express.Multer.File) {
    this.logger.log(PromoService.name);
    let image = null;
    const findOne = await promoEntity.findOne({ where: { name: body.name } });
    if (findOne) {
      throw new HttpException(
        {
          status: HttpStatus.BAD_REQUEST,
          message: 'Promo name already exists',
        },
        HttpStatus.BAD_REQUEST,
      );
    }
    if (file?.path) {
      image = `/assets${file.path.split('/assets')[1]}`;
    }
    const create = await promoEntity.create({ ...body, image, user_id });
    create.save();
    return {
      status: HttpStatus.CREATED,
      message: 'Promo has been create',
      create,
    };
  }

  async update(id: number, body: PromoField, file: Express.Multer.File) {
    this.logger.log(PromoService.name);
    const findOne = await promoEntity.findOne({ where: { id } });
    if (!findOne) {
      throw new HttpException(
        {
          status: HttpStatus.BAD_REQUEST,
          message: 'Promo not found',
        },
        HttpStatus.BAD_REQUEST,
      );
    }
    if (file?.path) {
      const image = `/assets${file.path.split('/assets')[1]}`;
      if (findOne.image) {
        try {
          removepath(`../..${findOne.image}`);
        } catch (err) {
          // empty
        }
      }
      findOne.image = image;
      findOne.save();
    }
    findOne.update(body, { where: { id } });
    return { status: HttpStatus.OK, message: 'Promo has been update' };
  }

  async destroy(id: number) {
    this.logger.log(PromoService.name);
    const findOne = await promoEntity.findOne({ where: { id } });
    if (!findOne) {
      throw new HttpException(
        {
          status: HttpStatus.BAD_REQUEST,
          message: 'Promo not found',
        },
        HttpStatus.BAD_REQUEST,
      );
    }
    if (findOne.image) {
      try {
        removepath(`../..${findOne.image}`);
      } catch (err) {
        // empty
      }
    }
    findOne.destroy();
    return { status: HttpStatus.OK, message: 'Promo has been delete' };
  }
}
