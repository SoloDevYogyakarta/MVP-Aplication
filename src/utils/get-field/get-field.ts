import { GetFieldType } from 'src/types/get-field.type';
import { readpath } from '../system/system';

export function getField(filename?: GetFieldType) {
  try {
    if (filename === 'fake') {
      return readpath(`../../database/dataTxt/${filename}.txt`);
    }
    return JSON.parse(readpath(`../../database/dataTxt/${filename}.txt`));
  } catch (err) {
    throw new Error(err);
  }
}
