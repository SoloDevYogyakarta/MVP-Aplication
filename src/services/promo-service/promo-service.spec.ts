import { faker } from '@faker-js/faker';
import { HttpStatus } from '@nestjs/common';
import { omit } from 'lodash';
import { createpath } from '../../utils/system/system';
import { getfield } from '../../utils/get-field/get-field';
import { PromoService } from './promo-service';

describe('PromoService', () => {
  let service: PromoService;
  let user_id!: number;

  beforeEach(() => {
    service = new PromoService();
  });

  it('should to be defined', () => expect(service).toBeDefined());

  it('render correctly', () => expect(service).toMatchSnapshot());

  try {
    user_id = getfield('user-http-entity').id;
  } catch (err) {
    // empty
  }

  if (user_id) {
    it('create', async () => {
      const result = await service.create(
        user_id,
        {
          name: faker.commerce.productName(),
          desc: faker.commerce.productDescription(),
          price: Number(faker.commerce.price()),
          discount: faker.number.int({ min: 1, max: 100 }),
          start_time: new Date(),
          end_time: new Date(),
        },
        {
          fieldname: 'file',
          originalname: '391282393_7054748857952078_2554999196306250130_n.jpg',
          encoding: '7bit',
          mimetype: 'image/jpeg',
          destination: '/Users/kenedy-/projects/mvpapplication/src/assets',
          filename: 'Lu_Fu10pTEdBhilu5Kjf6.jpeg',
          path: '/Users/kenedy-/projects/mvpapplication/src/assets/Lu_Fu10pTEdBhilu5Kjf6.jpeg',
          size: 409192,
        } as Express.Multer.File,
      );
      createpath(
        '../folder-text/promo-destroy-image-entity.txt',
        result.create,
      );
      expect(omit(result, ['create'])).toEqual({
        status: HttpStatus.CREATED,
        message: 'Promo has been create',
      });
    });
  }
});
