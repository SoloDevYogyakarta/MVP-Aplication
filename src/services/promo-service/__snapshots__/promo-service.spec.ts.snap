// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PromoService render correctly 1`] = `
PromoService {
  "logger": Logger {
    "context": "PromoService",
    "options": {},
  },
}
`;
